scripts/cmake/vcpkg_from_git.cmake

```
@@ -62,20 +65,30 @@ function(vcpkg_from_git)
     
 
     if(NOT DEFINED arg_OUT_SOURCE_PATH)
-        message(FATAL_ERROR "OUT_SOURCE_PATH must be specified.")
+        message(FATAL_ERROR "OUT_SOURCE_PATH must be specified")
     endif()
     if(NOT DEFINED arg_URL)
-        message(FATAL_ERROR "The git url must be specified")
+        message(FATAL_ERROR "URL must be specified")
     endif()
     if(NOT DEFINED arg_REF AND NOT DEFINED arg_HEAD_REF)
-        message(FATAL_ERROR "At least one of REF or HEAD_REF must be specified.")
+        message(FATAL_ERROR "At least one of REF or HEAD_REF must be specified")
+    endif()
+
+    vcpkg_list(SET git_fetch_shallow_param --depth 1)
+    if(DEFINED arg_FETCH_REF)
+        if(NOT DEFINED arg_REF)
+            message(FATAL_ERROR "REF must be specified if FETCH_REF is specified")
+        endif()
+        vcpkg_list(SET git_fetch_shallow_param)
+        set(ref_to_use "${arg_FETCH_REF}")
+    else()
+        set(ref_to_use "${arg_REF}")
     endif()
 
-    set(working_directory_param "")
-    set(ref_to_use "${arg_REF}")
+    vcpkg_list(SET working_directory_param)
     if(VCPKG_USE_HEAD_VERSION)
         if(DEFINED arg_HEAD_REF)
-            set(working_directory_param "WORKING_DIRECTORY" "${CURRENT_BUILDTREES_DIR}/src/head")
+            vcpkg_list(SET working_directory_param "WORKING_DIRECTORY" "${CURRENT_BUILDTREES_DIR}/src/head")
             set(ref_to_use "${arg_HEAD_REF}")
```

```suggestion
            set(ref_to_use "${arg_HEAD_REF}")
            vcpkg_list(SET git_fetch_shallow_param --depth 1)
```
We should not disable shallow fetch for `--head`.

Probably it makes the most sense to merge this `if()` with the above if/else on lines 78-88